### PR TITLE
Update testing READMEs and scripts

### PR DESCRIPTION
**Pull request by ben-harvey**
## Purpose

This change updates the top-level README and the Cypress test README to give more information on the types of tests contained in the QuickStart and how to run them. 

In the process, I did some cleanup and reorganization, including:
- standardize script names to use `snake_case` and the `.sh` suffix
- place scripts in their own `scripts` folder in cases where there are multiple scripts in a given folder
- standardize on `yarn` as a runner (vs `npm`) in `package.json` and READMEs
- reorganize the `tests` folder to remove nesting and duplicate `cypress.json` files
  - this included updates to the `tests/cypress` README, the `package.json`, and `cypress.json` itself
- refactor scripts to use the top-level `.github` directory for path references (rather than relative paths, which can break if the folder structure changes)
- remove:
  - commented-out code in testing files
  - scripts that reference dependencies that have since been removed (e.g. `nightwatch`)
 - add:
   -  error handling for scripts that require input but are called without it
   - explanatory comments for scripts
   - Cypress video and screenshot folders to `.gitignore`

Manual Testing: 

- ran all local scripts to make sure they work, including error cases
- ran the `deploy`, `deploySupport`, and `destroy` workflows to make sure that all path references were updated correctly
- ran any updated `package.json` scripts

#### Linked Issues to Close

_Links to issue(s) that are closed by this PR. Be sure to use the phrase "Closes #XXX" for each issue, so they automatically close_

## Approach

_How does this change address the issue?_

## Learning

_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Assorted Notes/Considerations

_List any other information that you think is important... a post-merge activity, someone to notify, etc._

#### Pull Request Creator Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [ ] Someone has been assigned this PR.
- [ ] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: scripts/dev.sh [hunk 1]
```diff
@@ -7,14 +7,19 @@ set -eu -o pipefail
...
+# move to the top level directory of the repo
+TOP_LEVEL_DIR="$(git rev-parse --show-toplevel)"
+cd "$TOP_LEVEL_DIR"
+
 # check node version
-if ! diff ../.nvmrc <(node -v) > /dev/null ; then
+if ! diff .nvmrc <(node -v) > /dev/null ; then
 	echo "Uh Oh! The current node version does not match the version required in .nvmrc"
 	echo "If you have installed nvm, simply running 'nvm use' in this directory should solve the problem"
 	echo "If you don't have nvm yet, the instructions in the README should sort you."
```
Comment (leslie-corbalt) [inline scripts/dev.sh:25]: what does `sort` mean?

File: .github/workflows/destroy.yml [hunk 2]
```diff
@@ -26,9 +26,9 @@ jobs:
           fi
       - uses: actions/checkout@v2
       - name: set branch specific variable names
-        run: ./.github/build_vars.sh set_names
+        run: ./.github/scripts/build_vars.sh.sh set_names
       - name: set variable values
-        run: ./.github/build_vars.sh set_values
+        run: ./.github/scripts/build_vars.sh.sh set_values
```
Comment (leslie-corbalt) [inline .github/workflows/destroy.yml:31]: `build_vars.sh.sh` ?

File: tests/cypress/README.MD [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+## Setup
+
+Run `yarn install` in this directory to install dependencies.
+
+## Running tests
+The `scripts` section of `package.json` defines the following scripts:
+  - `start`
+    - a wrapper to `dev.sh local`, which runs the application locally. Note that the `local` option for this script requires valid AWS credentials set per the [Serverless Framework docs](https://www.serverless.com/framework/docs/providers/aws/guide/credentials/)
+
+  - ` test`
```
Comment (leslie-corbalt) [inline tests/cypress/README.MD]: extra space after backtick

File: tests/cypress/README.MD [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+[Cypress](https://www.cypress.io/features) is an open source testing tool.
+
+## Setup
+
+Run `yarn install` in this directory to install dependencies.
+
+## Running tests
+The `scripts` section of `package.json` defines the following scripts:
+  - `start`
+    - a wrapper to `dev.sh local`, which runs the application locally. Note that the `local` option for this script requires valid AWS credentials set per the [Serverless Framework docs](https://www.serverless.com/framework/docs/providers/aws/guide/credentials/)
```
Comment (leslie-corbalt) [inline tests/cypress/README.MD]: `./dev.sh local` ?
If someone, without bash, tries to run `dev.sh local`, the script won't work (I don't think).

**Review by leslie-corbalt — COMMENTED** [review]
Looks good! I just had a few comments.

File: scripts/dev.sh (hunk 1 above)
Comment (ben-harvey) [reply scripts/dev.sh:25]: I think it's meant in the sense of 'sort out,' as in 'resolve a difficulty.'  But the object of 'sort' in that sense is usually the problem being sorted, not the person having the problem.  Looks like there is a British colloquial usage that takes the person as the object. 
In any case, I'll reword to clarify

File: .github/workflows/destroy.yml (hunk 2 above)
Comment (ben-harvey) [reply .github/workflows/destroy.yml:31]: Good catch, I could have sworn I fixed this

File: tests/cypress/README.MD (hunk 3 above)
Comment (ben-harvey) [reply tests/cypress/README.MD]: Fixed

File: tests/cypress/README.MD (hunk 4 above)
Comment (ben-harvey) [reply tests/cypress/README.MD]: I think the `.sh` extension just denotes that the file is a shell executable, but doesn't enforce which shell is used to run the file.  Mac uses `zsh` by default, and I can also run it on Mac using `sh` and `bash`.  Would you mind pulling the code and seeing if the script runs on Windows/Linux?

**Review by leslie-corbalt — APPROVED** [review]
LGTM!

File: scripts/dev.sh (hunk 1 above)
Comment (leslie-corbalt) [reply scripts/dev.sh]: :)

File: tests/cypress/README.MD (hunk 4 above)
Comment (leslie-corbalt) [reply tests/cypress/README.MD]: # 12:58:01 ~/git/macfc-quickstart bharvey-testing
$ cd scripts/
# 12:58:37 ~/git/macfc-quickstart/scripts bharvey-testing
$ dev.sh local
dev.sh: command not found
# 12:58:41 ~/git/macfc-quickstart/scripts bharvey-testing
$ ./dev.sh local
Uh Oh! The current node version does not match the version required in .nvmrc

**Pull request merged**